texts/0006-tauri-releases-schedule.md

```diff
@@ -0,0 +1,53 @@
+- Feature Name: tauri_release_schedule
+- Start Date: 2024-6-6
+- RFC PR: [tauri-apps/rfcs#14](https://github.com/tauri-apps/rfcs/pull/14)
+- Tracking Issue:
+
+# Summary
+
+Define a release schedule for tauri main repo, including crates, NPM packages and CLIs. 
+
+# Motivation
+
+As it stands now, there is no release schedule and releases are made either whenever one of the working group feels like it (mainly Lucas and myself (Amr)),
+or when there is a need for an urgent fix.
+
+As a tauri user or contributor, there is no indication when a release might happen and if will it include a certain feature
+one might be waiting for which leads me to either try to figure how to use unreleased features directly from source or wait indefinitely.   
+
+A release schedule should make the releasing process more predictable for maintainers and users and prevents the tauri project from ever appearing stagnant.
+
+# Explanation
+
+We follow semver versioning for all tauri crates, NPM packages and CLIs and so we have 6 type of releases we are ever gonna make for stable releases,
+here is what each release type cycle would look like:
+
+- `patch`: weekly
+  - `patch` versions can be released when needed, in the case of an urgent bug fix or security vulnerability. 
+  - PRs with `minor` bumps shall be marked as `Ready for next minor` and will not be merged until the next `minor` release cycle.
+    This policy exists to ease the release of `patch` versions without cherrypicking into a separate branch and releasing from there.    
+- `minor`: every 2 months
+- `major`: yearly
+- `alpha`: weekly
+- `beta`: weekly
+- `rc`: weekly
+
+#### When to start alpha for next major?
+
+- `alpha` releases should start 3 months before the next planned `major` release date and should last 1 months or 2 until somewhat stable and ready for beta.
+- `beta` releases last until the security audit is done
+- `rc` releases last until almost every major bug is squashed and until the next planned `major` release date. `rc` can last after that date if there is more bugs to fix but should not exceed one month after.
+
+## Security Considerations
+
+Security is one of our top priorities for tauri and we have audited each major and minor release up until now, and we plan on keeping that. However, security audits usually takes 
+a fair bit of time and auditing each `minor` release could take a while and might not meet the release date of that minor.
+
+Right now, we have been publishing `minor` releases without waiting for audit to finish but auditing will continue and if secuirty issues are found,
+we fix and release them in a `patch` release afterwards. Same process shall be kept as is going forward.
+
+Major releases will never be published before its audit is done and findings are fixed.
+
+# Drawbacks
+
+Minor releases maybe released before its security audit is over but that's the same as the current process.
```

Comment:
Maybe there could be some way to indicate the latest fully audited minor release to mitigate that?

Reply:
Yeah that's a good idea, something like a status page in https://tauri.app or maybe as a project in https://github.com/tauri-apps/tauri/projects

Reply:
while that might be possible for NPM dependencies, crates.io doesn't have this feature for the Rust crates.